### ISSUE TITLE
Unable to install plugin-online-media

### ISSUE THREAD

**Issue by low-batt** [iina/iina#4411]
**System and IINA version:**

- macOS 13.3.1 (a)
- IINA `develop` [ee48143](https://github.com/iina/iina/commit/ee48143970f683de1861b6ab8d480043dd0f17b4)

**Expected behavior:**
The IINA plugin `plugin-online-media` can be installed from GitHub.

**Actual behavior:**
Installation fails reporting the following error:
![iina-plugin-install-fail](https://github.com/iina/iina/assets/86170219/58e77a81-ba07-4e33-abd3-8dea096e18f2)

<details>
<summary><b>IINA log:</b></summary>

```
14:58:57.207 [iina][d] Installing plugin from GitHub URL: https://github.com/iina/plugin-online-media
14:59:02.164 [iina][d] Loading JS plugin from /Users/low-batt/Library/Application Support/com.colliderli.iina/plugins/.temp.C8BB9BA7-9FBB-4F2A-87A7-0F7BBAD6B494
14:59:02.165 [iina][d] The plugin Online Media  doesn't contain a ghVersion field or its ghRepo doesn't match the current requested URL https://github.com/iina/plugin-online-media.
```
</details>

<br>

**Steps to reproduce:**
Run the following command in [Terminal](https://support.apple.com/guide/terminal/welcome/mac) to enable the plugin system:
```bash
defaults write com.colliderli.iina iinaEnablePluginSystem true
```

- Start IINA
- Click on  `Settings…` under the `IINA` menu
- The settings panel appears
- On the left side of the panel click on `Plugins`
- On the right side of the panel click on `Install from GitHub…`
- The popup `Install a new plugin from GitHub` appears
- In the popup click on `iina/plugin-online-media`
- The `Install` button activates
- Click on `Install`
- The error popup shown above appears

- [x] MPV does not have this problem.

**How often does this happen?**
Every time.

**Comment by low-batt** [comment]
I cloned [plugin-online-media](https://github.com/iina/plugin-online-media) and created a zip file with the extension iinaplgz. I then used the `Install from a local package…` button to install the plugin. The installation completed without error. Installed this way the plugin seems to be working:

```text
15:25:37.832 [player0 - Online Media ][d] ytdl:// hook
15:25:37.855 [player0][d] Started auto load
15:25:37.857 [fmatcher][d] **Start matching
15:25:44.203 [player0 - Online Media ][d] ytdl full hook
15:25:47.186 [player0][d] Track list changed
15:25:47.188 [player0 - Online Media ][d] ytdl preload hook
15:25:47.191 [player0][d] File loaded
```

The extra blank space in the log messages needs to be corrected.

**Comment by smishralm10** [comment]
> I cloned [plugin-online-media](https://github.com/iina/plugin-online-media) and created a zip file with the extension iinaplgz.

How did you create a zip file with `iinaplgz` extension? I tried but couldn't get it to work.

While investigating, I found that `plugin-online-media` doesn't contain ghVersion and ghRepo in its `Info.json` which it expects. Having said that, if that is the case, it shouldn't work while using `Install from a local package..` either  as they both use `JavascriptPlugin(filename: _)` to create the plugin. 

I need a little help to install the plugin using `Install from local package...`

**Comment by low-batt** [comment]
I'm not the authority on the plugin system, so I'm unsure of the official way to bundle a plugin. This is how I did it.

Start [Terminal](https://support.apple.com/guide/terminal/welcome/mac). Navigate to the git clone of `plugin-online-media`. In that directory run the following command:
```bash
zip -r plugin-online-media.iinaplgz *
```

Output from a terminal session running that command below.

I confirmed `Install from local package…` is able to install from that file.

<details>
<summary><b>Terminal session:</b></summary>

```text
low-batt@gag plugin-online-media (main %=)$ pwd
/Users/low-batt/Documents/builds/plugin-online-media/plugin-online-media
low-batt@gag plugin-online-media (main=)$ zip -r plugin-online-media.iinaplgz *
  adding: Info.json (deflated 47%)
  adding: README.md (deflated 39%)
  adding: dist/ (stored 0%)
  adding: dist/global.js (deflated 57%)
  adding: dist/index.js (deflated 60%)
  adding: dist/downloads-window.js (deflated 59%)
  adding: downloads.html (deflated 57%)
  adding: package-lock.json (deflated 76%)
  adding: package.json (deflated 59%)
  adding: pref.html (deflated 70%)
  adding: src/ (stored 0%)
  adding: src/ytdl-hook.ts (deflated 65%)
  adding: src/utils.ts (deflated 60%)
  adding: src/ytdl.d.ts (deflated 68%)
  adding: src/download.ts (deflated 65%)
  adding: src/downloads-window.js (deflated 59%)
  adding: src/global.ts (deflated 64%)
  adding: src/index.ts (deflated 62%)
  adding: src/add-video.ts (deflated 68%)
  adding: src/binary.ts (deflated 45%)
  adding: src/options.ts (deflated 60%)
  adding: tsconfig.json (deflated 39%)
low-batt@gag plugin-online-media (main %=)$ unzip -l plugin-online-media.iinaplgz 
Archive:  plugin-online-media.iinaplgz
  Length      Date    Time    Name
---------  ---------- -----   ----
      756  05-12-2023 17:23   Info.json
      614  05-12-2023 17:23   README.md
        0  05-12-2023 17:23   dist/
     5648  05-12-2023 17:23   dist/global.js
    10119  05-12-2023 17:23   dist/index.js
     7995  05-12-2023 17:23   dist/downloads-window.js
     1340  05-12-2023 17:23   downloads.html
   165605  05-12-2023 17:23   package-lock.json
      858  05-12-2023 17:23   package.json
     3227  05-12-2023 17:23   pref.html
        0  05-12-2023 17:23   src/
     5831  05-12-2023 17:23   src/ytdl-hook.ts
     4483  05-12-2023 17:23   src/utils.ts
     1985  05-12-2023 17:23   src/ytdl.d.ts
     3352  05-12-2023 17:23   src/download.ts
     1785  05-12-2023 17:23   src/downloads-window.js
     3678  05-12-2023 17:23   src/global.ts
     1177  05-12-2023 17:23   src/index.ts
     7832  05-12-2023 17:23   src/add-video.ts
      666  05-12-2023 17:23   src/binary.ts
     1333  05-12-2023 17:23   src/options.ts
      265  05-12-2023 17:23   tsconfig.json
---------                     -------
   228549                     22 files
low-batt@gag plugin-online-media (main %=)$ 
```
</details>

**Comment by smishralm10** [comment]
@low-batt Thanks for your help, I was able to reproduce the issue.

I noticed, while using `Install from local package` it makes a call to `create(fromPackageUrl: _) ` method which do not have a guard statement checking for `ghVersion` and  `githubURLString`(created using ghRepo). This isn't the case when creating plugin object using `create(fromGitURL: _)` method. 

One solution would be to add these two fields(ghVersion and ghRepo) in `Info.json` file of `plugin-online-media` just like in `plugin-userscript` and make it a standard for other plugins as well for future. Another solution would be to remove the guard statement altogether. The later could also be a valid solution because the only purpose of the guard statement, as far as I know, is to check whether the plugin is from valid source which is not done when installing plugin using `Install from local package..`
